packages/sdk/src/entities/bucket.ts

```diff
@@ -566,10 +648,18 @@ export class BucketManager {
       return { result };
     } catch (error: unknown) {
       if (error instanceof ContractFunctionExecutionError) {
-        // TODO: We're optimistically assuming an error means the bucket doesn't exist
-        // 00: t0134 (method 3844450837) -- contract reverted (33)
-        // 01: t0134 (method 6) -- contract reverted (33)
-        throw new BucketNotFound(bucket);
+        if (error.message.includes("contract reverted")) {
+          const isOutOfGasError =
+            error.message.includes("wasm `unreachable` instruction executed") ||
```

Comment:
meta observation: the fvm/wasm errors can be a bit too ambiguous. we'll see something like this if you query a bucket and it doesnt exist on the network, which is non-descriptive:
```
00: t0134 (method 3844450837) -- contract reverted (33)
01: t0134 (method 6) -- contract reverted (33)
```
but an out of gas issue will have an extra line that at least helps a bit:
```
02: t0134 (method 6) -- out of gas (7)
```
i.e., the js client has to make some assumption to help handle these errors nicely for the dev.